File: src/main/java/com/tobe/healthy/push/presentation/PushCommandController.java
package com.tobe.healthy.push.presentation;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.tobe.healthy.ApiResult;
import com.tobe.healthy.config.security.CustomMemberDetails;
import com.tobe.healthy.push.application.PushCommandService;
import com.tobe.healthy.push.presentation.dto.in.CommandRegisterToken;
import com.tobe.healthy.push.presentation.dto.in.CommandRegisterTokenWithWebView;
import com.tobe.healthy.push.presentation.dto.in.CommandSendPushAlarm;
import com.tobe.healthy.push.presentation.dto.in.CommandSendPushAlarmToMember;
import com.tobe.healthy.push.presentation.dto.out.CommandRegisterTokenResult;
import com.tobe.healthy.push.presentation.dto.out.CommandSendPushAlarmResult;

import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/api/v1/push")
@RequiredArgsConstructor
public class PushCommandController {

	private final PushCommandService pushCommandService;

	@PostMapping
	public ApiResult<CommandRegisterTokenResult> registerFcmToken(
		@RequestBody CommandRegisterToken request,
		@AuthenticationPrincipal CustomMemberDetails member) {
		return ApiResult.success("토큰을 저장하였습니다.", pushCommandService.registerFcmToken(request, member.getMemberId()));
	}

	@PostMapping("/webview")
	public void registerFcmTokenWithWebView(@RequestBody CommandRegisterTokenWithWebView request) {
		pushCommandService.registerFcmTokenWithWebView(request);
	}

	@PostMapping("/send")
	public ApiResult<CommandSendPushAlarmResult> sendPushAlarm(
		@RequestBody CommandSendPushAlarm request) {
		return ApiResult.success("푸시 전송에 성공하였습니다.", pushCommandService.sendPushAlarm(request));
	}

	@PostMapping("/{memberId}")
	public ApiResult<CommandSendPushAlarmResult> sendPushAlarm(
		@PathVariable Long memberId,
		@RequestBody CommandSendPushAlarmToMember request) {
		return ApiResult.success("푸시 전송에 성공하였습니다.", pushCommandService.sendPushAlarm(memberId, request));
	}
}

File: src/main/kotlin/com/tobe/healthy/push/presentation/PushCommandController.kt
package com.tobe.healthy.push.presentation

import com.tobe.healthy.ApiResultResponse
import com.tobe.healthy.config.security.CustomMemberDetails
import com.tobe.healthy.push.application.PushCommandService
import com.tobe.healthy.push.domain.dto.`in`.CommandRegisterToken
import com.tobe.healthy.push.domain.dto.`in`.CommandRegisterTokenWithWebView
import com.tobe.healthy.push.domain.dto.`in`.CommandSendPushAlarm
import com.tobe.healthy.push.domain.dto.`in`.CommandSendPushAlarmToMember
import com.tobe.healthy.push.domain.dto.out.CommandRegisterTokenResult
import com.tobe.healthy.push.domain.dto.out.CommandSendPushAlarmResult
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/push/v1")
class PushCommandController(
    private val pushCommandService: PushCommandService,
) {

    @PostMapping
    fun registerFcmToken(
        @RequestBody request: CommandRegisterToken,
        @AuthenticationPrincipal member: CustomMemberDetails
    ): ApiResultResponse<CommandRegisterTokenResult> {
        return ApiResultResponse(
            message = "토큰을 저장하였습니다.",
            data = pushCommandService.registerFcmToken(request, member.memberId)
        )
    }

    @PostMapping("/webview")
    fun registerFcmTokenWithWebView(@RequestBody request: CommandRegisterTokenWithWebView) {
        pushCommandService.registerFcmTokenWithWebView(request)
    }

    @PostMapping("/send")
    fun sendPushAlarm(
        @RequestBody request: CommandSendPushAlarm
    ): ApiResultResponse<CommandSendPushAlarmResult> {
        return ApiResultResponse(
            message = "푸시 전송에 성공하였습니다.",
            data = pushCommandService.sendPushAlarm(request)
        )
    }

    @PostMapping("/{memberId}")
    fun sendPushAlarm(
        @PathVariable memberId: Long,
        @RequestBody request: CommandSendPushAlarmToMember
    ): ApiResultResponse<CommandSendPushAlarmResult> {
        return ApiResultResponse(
            message = "푸시 전송에 성공하였습니다.",
            data = pushCommandService.sendPushAlarm(memberId, request)
        )
    }
}
